출발시간페이지

Yunjigong opened the pull request:
provider는 아직 공부 중이어서 작업하지 못했습니다. 일단 api에 있는 데이터를 가져왔을 때 데이터를 수정하지 않고 사용할 수 있도록 코드를 조금 수정했습니다. 그리고 bus_direction.dart파일은 제가 출발시간페이지 수정하기 전 코드가 있는데 실수로 같이 올라간 것 같아서 삭제했습니다.

yunji420 committed:
08-16

provider는 아직 공부 중이어서 작업하지 못했습니다. 일단 api에 있는 데이터를 가져왔을 때 데이터를 수정하지 않고 사용할 수 있도록 코드를 조금 수정했습니다. 그리고 bus_direction.dart파일은 제가 출발시간페이지 수정하기 전 코드가 있는데 실수로 같이 올라간 것 같아서 삭제했습니다.

yunji420 committed:
08-22

폴더에 있는 모델 부분과 연결이 안되는 것 같고 그래서 제대로 작동하는지 확인해보지 못했습니다. 프로바이더를 설치했는데 changenotifier가 작동하지 않아 찾아보고 있는 중입니다.

yunji420 committed:
0827

classes.dart 파일에 landmarkAliases 부분에 id가 없어서 추가했습니다.
busTimeData 파일 카멜케이스 수정했고 dynamic 수정했습니다.
근데 필요한 리스트를 만들어내지 못해서 수정이 조금 필요한 상태입니다.

yunji420 committed:
08-31

ghost commented:
출발시간 페이지하다가 두가지 문제가 있었는데
첫번째는 list만드는 부분을 확인하는 방법을 몰라서 필요한 리스트를 만든 건지 확인을 못했고
(여러번 확인해보기는 했어요!)
두번째는 provider.of를 사용하는 방법을 모르겠습니다. 
두가지 문제로 출발시간페이지 부분 작업이 막혀서 상세페이지 부분도 같은 정도까지 진행했습니다.

junwha commented:
주말 내로 리뷰할게요!

junwha committed:
fix: dependency

junwha committed:
fix: resolve dependency problems

junwha committed:
refactor: rename with convention

junwha commented on lib/busTimeData.dart:
```diff
@@ -0,0 +1,123 @@
+//   return routeKey;
+// }
+
+deparatTime() async {
```
Return type 모두 명확하게 명시해주세요! dynamic은 가독성을 저하시킵니다

junwha commented on lib/busTimeData.dart:
```diff
@@ -0,0 +1,123 @@
+  List<UlsanBusTimeTables> timeTable = await getAPIUlsanBusTimeTables();
+  List<LaneToTracks> laneToTracks = await getAPILaneToTracks();
+  List<String> busName_list = [];
+  for (int i = 0; i < timeTable.length; i++) {
```
여긴 잘했어요!

junwha commented on lib/busTimeData.dart:
```diff
@@ -0,0 +1,123 @@
+  List<Map<String, String>> _busTime = [];
+  for (int i = 0; i <= timeTable.length; i++) {
+    Map<String, String> data = {};
+    data["bus"] = bus[i];
```
Map이 아닌 클래스를 생성해서 담아주세요!

junwha reviewed (changes requested):
전체적으로 깔끔하게 잘 짰는데 코드 컨벤션과 클래스 사용만 주의하면 될 것 같아요:)

junwha commented on lib/bustime.dart:
```diff
@@ -10,119 +11,57 @@ class BusTime extends StatefulWidget {
       TimeOfDay listTime = TimeOfDay(
-          hour: int.parse(_busTime[i]['time'].split(":")[0]),
-          minute: int.parse(_busTime[i]['time'].split(":")[1]));
+          hour: int.parse(_busTime[i]['time'].substring(0, 2)),
```
이런 처리들은 가능하면 프로바이더 쪽에 분리시키는게 좋을 것 같아요

junwha commented:
아 그리고 PM 시간대에 버스가 안 뜨는 문제가 있는 것 같아요!

ghost commented:
> 아 그리고 PM 시간대에 버스가 안 뜨는 문제가 있는 것 같아요!

아직 provider를 연결 못해서 busTime 뷰 구현한 파일에서 처음에 했던 것처럼 몇가지 데이터로 리스트 만들어 둔 걸로 연결되어 있어서 그런 것 같습니다...!

yunji420 committed:
09-19

provider는 아직 구현하지 못했습니다. Map이 아니라 class로 담으라고 하신부분은 일단 담아보기는 했는데 이게 맞는지는 모르겠습니다... time을 hour과 min으로 바꾸는 부분을 추가했습니다. detailedPageData 부분에 불필요한부분 수정했습니다.

맡은부분 끝까지 하지 못해서 죄송합니다

yunji420 committed:
Merge branch 'dpt_page' of https://github.com/junwha0511/Bus-HeXA-App into dpt_page
